datus/prompts/prompt_manager.py

```
@@ -32,20 +32,36 @@ def __init__(self):
+            return self.user_templates_dir
         else:
-            self.templates_dir = self.default_templates_dir
-            logger.info(f"Using default template directory: {self.default_templates_dir}")
+            return self.default_templates_dir
 
-        self._env = Environment(loader=FileSystemLoader(str(self.templates_dir)), trim_blocks=True, lstrip_blocks=True)
+    def _get_env(self) -> Environment:
+        """Get Jinja2 environment, creating or updating if needed."""
+        current_templates_dir = self.templates_dir
+        if self._env is None or self._last_templates_dir != current_templates_dir:
+            self._env = Environment(
```

Ruff (0.14.3)

By default, jinja2 sets `autoescape` to `False`. Consider using `autoescape=True` or the `select_autoescape` function to mitigate XSS vulnerabilities.

(S701)